parsing/parser.mly

```diff
@@ -1816,6 +1810,8 @@ simple_pattern_not_ident:
       { $1 }
   | mod_longident DOT simple_delimited_pattern
       { mkpat @@ Ppat_open(mkrhs $1 1, $3) }
+  | mod_longident DOT LBRACKET RBRACKET
+    { mkpat @@ Ppat_construct ( mkrhs (Lident "[]") 1, None) }
```

Comment:
Thanks for the comments.  I intentionally kept the behaviour the same as trunk, where `M` is ignored in the pattern `M.[]`.  But I agree that using `M` to resolve `[]` is better, so I've added `Ppat_open` in 1c4ee3a.